mesonbuild/msetup.py

```diff
@@ -169,16 +169,18 @@ def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) ->
         has_partial_build = os.path.isdir(priv_dir)
         if has_valid_build:
             if not reconfigure and not wipe:
-                print('Directory already configured.\n'
-                      '\nJust run your build command (e.g. ninja) and Meson will regenerate as necessary.\n'
+                print('Directory already configured.\n\n'
+                      'Just run your build command (e.g. ninja) and Meson will regenerate as necessary.\n'
                       'If ninja fails, run "ninja reconfigure" or "meson setup --reconfigure"\n'
-                      'to force Meson to regenerate.\n'
-                      '\nIf build failures persist, run "meson setup --wipe" to rebuild from scratch\n'
-                      'using the same options as passed when configuring the build.'
-                      '\nTo change option values, run "meson configure" instead.')
+                      'to force Meson to regenerate.\n\n'
+                      'If build failures persist, run "meson setup --wipe" to rebuild from scratch\n'
+                      'using the same options as passed when configuring the build.\n'
+                      'To change option values, run "meson configure" instead.')
+                # This is not an error for backward compatibility:
+                # https://github.com/mesonbuild/meson/pull/4249
```

Comment:
IMHO the comment is wrong -- this is not backwards compat, just sensible behavior.

Independent of what we do when rerunning "setup" and having it do nothing, you do have another PR to make it do something if `--options` are passed, which is another perhaps reasonable thing to do here... but at the core of it I still argue that this is "intended UX", not "preserved for backwards compat".

Reply:
It should be hard error or set new options, that can be argued either way. But current behaviour of ignoring options is totally wrong. I wrote that comment when I thought we should be heading toward hard error, but I changed my mind since then. I'll put more nuance into that comment, indeed.

Reply:
The nuance is... better... although as I said, I still disagree with your "should either", but whatever. :P  

However, the update is in the wrong commit, currently you say one thing in the first commit and then change your mind in the unrelated second commit!

Also, the commit message for the second commit (still) should be fixed to state what it's actually doing now.

Reply:
done